lib/middleware/connect.js

```diff
@@ -1,24 +1,30 @@
 'use strict';
 
 var Raven = require('../client');
+var utils = require('../utils');
 
 // Legacy support
 var connectMiddleware = function (client) {
   return connectMiddleware.errorHandler(client);
 };
 
+var getClient = function (clientOrDSN) {
+  // Raven is an instance, so use Raven.constructor for instanceof check
+  return clientOrDSN instanceof Raven.constructor ? clientOrDSN : new Raven.Client(clientOrDSN);
+};
+
 // Error handler. This should be the last item listed in middleware, but
 // before any other error handlers.
-connectMiddleware.errorHandler = function (client) {
-  client = client instanceof Raven.Client ? client : new Raven.Client(client);
-  return client.errorHandler();
+connectMiddleware.errorHandler = function (clientOrDSN) {
+  utils.consoleAlert('top-level Raven.middleware.* methods have been deprecated and will be removed in v2.0; use instance methods instead');
```

Comment:
Good explanation – I agree. My fear is that people will see that message and have no idea what you're talking about, especially since it's not yet reflected in docs.

Can we do something like:

```javascript
utils.consoleAlert('top-level Raven.middleware.* methods have been deprecated and will be removed in v2.0; use client.requestHandler()');
```